### PR TITLE
ci: use github actions

### PR DESCRIPTION
**Pull request by erisu**
### Motivation, Context & Description

Drop Travis CI and AppVeyor for GitHub Actions

* Drop travis ci
* Drop appveyor
* Update coverage to include e2e tests
* Update test scripts

### Testing

- `npm t`
- GitHub Actions Run

### Checklist

- [x] I've run the tests to see all new and existing tests pass

### CODE REVIEW

**Comment by timbru31** [comment]
Is there any way we can see if the actions will succeed once merged?

File: package-lock.json [hunk 1]
```diff
@@ -0,0 +1,3509 @@
+{
```
Comment (timbru31) [inline package-lock.json:1]: This file should be added in a separate PR, IMO

**Comment by NiklasMerz** [comment]
> Is there any way we can see if the actions will succeed once merged?

Looks like we can see the results in the fork. Seems good https://github.com/erisu/cordova-ios/actions

File: package-lock.json (hunk 1 above)
Comment (erisu) [reply package-lock.json:1]: Removed from commit but still exists because of the base branch.

**Comment by erisu** [comment]
@timbru31 here is an example last run after the recent rebase: https://github.com/erisu/cordova-ios/actions/runs/111508020

**Comment by erisu** [comment]
I cleaned up the test changes and here is the latest CI run results:

https://github.com/erisu/cordova-ios/actions/runs/116486160
